src/gleam/dynamic.gleam

```diff
@@ -322,9 +351,12 @@ pub fn list(
 ) -> Decoder(List(inner)) {
   fn(dynamic) {
     use list <- result.try(shallow_list(dynamic))
-    list
-    |> list.try_map(decoder_type)
-    |> map_errors(push_path(_, "*"))
+    let result = try_map_with_index(list, decoder_type)
+    case result {
+      Ok(values) -> Ok(values)
+      Error(#(index, errors)) ->
+        Error(list.map(errors, push_path(_, int.to_string(index))))
+    }
```

Comment:
Let's make the helper function specific 
```suggestion
    let result = decode_list(list, decoder_type, 0, [])
    case result {
      Ok(values) -> Ok(values)
      Error(#(index, errors)) ->
        Error(list.map(errors, push_path(_, index)))
    }
```

Reply:
@lpil do you think it's alright as it is now?